experimental/dds/tree2/src/feature-libraries/editable-tree-2/README.md

```diff
@@ -0,0 +1,84 @@
+        This requires some extra properties to prevent MapNodes and sequences from blocking the traversal, and provides guidance for which properties should be enumerable own properties.
+    -   Adapters can be provided to (lazily) view any subtree in a more JavaScript object like form, with different options to pick different trad-offs like boxed vs. unboxed and including getters (as non-enumerable properties) back to the editable view. (TODO: actually implement this):
+
+    Like with the generic tree readers above, this case intentionally does not provide editing.
+    Also like the generic tree readers case, it provides an easy way for code which does understand the schema to get back to the Schema Aware API for editing if needed.
+    (TODO: Add APIs for doing such casts safely.
+    Possible way to do instance of checks or symbols to recover which objects are nodes and fields)
+
+-   Subscribing to changes.
+    This includes cases like updating the user interface to reflect changes
+
+    This could be considered part of either reading API, but is being addressed separately here.
+    APIs for receiving these changes take the form of registering callbacks for events.
+    These events can be triggered by anything that impacts the content of the tree including local edits, remote edits, merges, rebasing local edits, transaction rollback, undo etc.
+    These can also be batching together, possibly eliminating intermediate states and reordering operations that don't impact each-other (like edits to different parts of the tree).
+    This means these event subscriptions should only be treated as changes to the content, and users of them should not attempt to ascribe meaning or providence to any particular event.
+
+    Some control over these events is available via branching which provides snapshot isolation (among other things):
+    this is done at a higher level than the tree API covered here but can be accessed via each tree entities context property. (TODO: provide this access).
+
+## Javascript Object API: `enumerable` and `own` properties
+
+See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties for context.
+Note that `enumerable` has nothing to do with iterable: Iterable is not relevant to this section.
+
+Despite this tree API primarily being a TypeScript API, and TypeScript types having no way to indicate if members are `enumerable` or `own`, these details matter, even to TypeScript users.
+
+For example, TypeScript assumes all members of interfaces are `enumerable` and `own` when using the [Object spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) operator.
+Typescript also assumes all non-method members of class are `enumerable` and `own`, but allows assigning the class to an interface where it will start treating the methods as `enumerable` and `own` despite it being the same object.
```

Comment:
```suggestion
Typescript also assumes all non-method members of classes are `enumerable` and `own`, but allows assigning the class to an interface where it will start treating the methods as `enumerable` and `own` despite it being the same object.
```